ModLoader/Dev Console/ConsoleInput.cs

```diff
@@ -0,0 +1,94 @@
+using UnityEngine;
+using System;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.IO;
+
+namespace Windows
+{
+	public class ConsoleInput
+	{
+		//public delegate void InputText( string strInput );
+		public event System.Action<string> OnInputText;
+		public string inputString;
+
+		public void ClearLine()
+		{
+			Console.CursorLeft = 0;
+			Console.Write( new String( ' ', Console.BufferWidth ) );
+			Console.CursorTop--;
+			Console.CursorLeft = 0;
+		}
+
+		public void RedrawInputLine()
+		{
+			if ( inputString.Length == 0 ) return;
+
+			if ( Console.CursorLeft > 0 )
+				ClearLine();
+
+			System.Console.ForegroundColor = ConsoleColor.Green;
+			System.Console.Write( inputString );
+		}
+
+		internal void OnBackspace()
+		{
+			if ( inputString.Length < 1 ) return;
+
+			inputString = inputString.Substring( 0, inputString.Length - 1 );
+			RedrawInputLine();
+		}
+
+		internal void OnEscape()
+		{
+			ClearLine();
+			inputString = "";
+		}
+
+		internal void OnEnter()
+		{
+			ClearLine();
+			System.Console.ForegroundColor = ConsoleColor.Green;
+			System.Console.WriteLine( "> " + inputString );
+
+			var strtext = inputString;
+			inputString = "";
+
+			if ( OnInputText != null )
+			{
+				OnInputText( strtext );
+			}
+		}
+
+		public void Update()
+		{
+			if ( !Console.KeyAvailable ) return;
+			var key = Console.ReadKey();
+
```

Comment:
These if statements could be cleaned up a bit with a switch statements but it's not necessary